core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -5609,7 +5609,22 @@ protected function filterMenu($intFilterPanel)
 			// Limit the options if there are root records
 			if (isset($GLOBALS['TL_DCA'][$table]['list']['sorting']['root']) && $GLOBALS['TL_DCA'][$table]['list']['sorting']['root'] !== false)
 			{
-				$arrProcedure[] = "id IN(" . implode(',', $this->Database->getChildRecords($GLOBALS['TL_DCA'][$table]['list']['sorting']['root'], $this->strTable)) . ")";
+				$rootIds = array_map('\intval', $GLOBALS['TL_DCA'][$table]['list']['sorting']['root']);
+
+				// Also add the child records of the table (see #1811)
+				if ($GLOBALS['TL_DCA'][$table]['list']['sorting']['mode'] == 5)
```

Comment:
This should be done in mode 6 as well.

Reply:
No, this is fine the way it is. By using `$table` instead of `$this->strTable`, we are already accounting for mode 6.

Reply:
what if the parent table of the current is mode 6? Not sure if that would even work though 😆

Reply:
Can you provide a test case to reproduce? Otherwise we should probably merge the PR and see if there are any edge cases, because it is blocking the release of Contao 4.4.51.